src/content/docs/2/guide/plugins/index.mdx

```diff
@@ -2,12 +2,301 @@
+    .setup(|app, api| {
+      let timeout = api.config.timeout;
+      Ok(())
+    })
+    .build()
+}
+```
+
+In the Tauri application where the plugin is used, the configuration is set on `tauri.conf.json`:
+
+```json
+{
+  "build": { ... },
+  "tauri": { ... },
+  "plugins": {
+    "plugin-name": {
+      "timeout": 30
+    }
+  }
+}
+```
+
+Where `plugin-name` is your actual plugin name.
+
+### Lifecycle Events
+
+Plugins can hook into several lifecycle events:
+
+- [setup](#setup): Plugin is being initialized
+- [on_navigation](#on_navigation): Webview is attempting to perform a navigation
+- [on_webview_ready](#on_webview_ready): New window is being created
+- [on_event](#on_event): Event loop events
+- [on_drop](#on_drop): Plugin deconstruction
+
+#### setup
+
+- **When**: Plugin is being initialized
+- **Why**: Register mobile plugins, manage state, run background tasks
+
+```rust
+use tauri::plugin::Builder;
+
+Builder::new("<plugin-name>")
+  .setup(|app, api| {
+
+    // TODO: some nice example code here?
+
+    Ok(())
+  })
+```
+
+#### on_navigation
+
+- **When**: Webview is attempting to perform a navigation
+- **Why**: TODO:
+
+{/* TODO: Why */}
+
+Returning `false` cancels the navigation.
+
+```rust
+use tauri::plugin::Builder;
+
+Builder::new("<plugin-name>")
+  .on_navigation(|window, url| {
+    println!("window {} is navigating to {}", window.label(), url);
+    // Cancels the navigation if forbidden
+    url.scheme() != "forbidden"
+  })
+```
+
+#### on_webview_ready
+
+- **When**: New window is being created
+- **Why**: TODO:
+
+{/* TODO: Why */}
+
+```rust
+use tauri::plugin::Builder;
+
+Builder::new("<plugin-name>")
+  .on_webview_ready(|window| {
+    // TODO: Maybe a more practical example?
+    println!("window {} created", window.label());
+  })
+```
+
+#### on_event
+
+- **When**: Event loop events
+- **Why**: TODO:
+
+{/* TODO: Why */}
+
+With this lifecycle hook you can be notified of any event loop [events](https://docs.rs/tauri/2.0.0-alpha/tauri/enum.RunEvent.html).
+
+```rust
+use tauri::{plugin::Builder, RunEvent};
+
+Builder::new("<plugin-name>")
+  .on_event(|app, event| {
+    match event {
+      RunEvent::ExitRequested { api, .. } => {
+        // TODO: Fill out examples
+        // user requested a window to be closed
+      }
+      RunEvent::Exit => {
+        // app is going to exit, you can cleanup here
+      }
+      _ => {}
+    }
+  })
+```
+
+#### on_drop
+
+- **When**: Plugin deconstruction
+- **Why**: TODO:
+
+{/* TODO: Why */}
+
+See [`Drop`](https://doc.rust-lang.org/std/ops/trait.Drop.html) for more information.
+
+```rust
+use tauri::plugin::Builder;
+
+Builder::new("<plugin-name>")
+  .on_drop(|app| {
+    // TODO: Code example
+    // plugin has been destroyed...
+  })
+```
+
+### Adding Commands
+
+{/* TODO: Could this be linked to a more generic commands guide and only cover plugin-specific things here? */}
+{/* What here is plugin-development specific vs. more general? */}
```

Comment:
Would love some help on this one

Reply:
Most of this is plugin specific, but it would be helpful to link to the fully fledged commands guide

Reply:
Is there anything here that would go in that more full-fledged command guide? Might be good to go ahead and pull those out as a stub page and link to it so we don't need to refactor it later